R/appMetadata.R

```diff
@@ -98,10 +98,18 @@ checkAppLayout <- function(appDir, appPrimaryDoc = NULL) {
   # can be run/served.
 }
 
-# infer the mode of the application from files in the root dir
-inferAppMode <- function(absoluteRootFiles,
-                         usesQuarto = NA,
-                         isShinyappsServer = FALSE) {
+# Infer the mode of the application from included files. Most content types
+# only consider files at the directory root. TensorFlow saved models may be
+# anywhere in the hierarchy.
+inferAppMode <- function(
+  appDir,
+  appFiles,
+  usesQuarto = NA,
+  isShinyappsServer = FALSE) {
+
+  rootFiles <- appFiles[dirname(appFiles) == "."]
+  absoluteRootFiles <- file.path(appDir, rootFiles)
+  absoluteAppFiles <- file.path(appDir, appFiles)
```

Comment:
Weird that TensorFlow files can be anywhere in the hierarchy! I guess for those, you may be using R in any working dir and are just pointing to a TF file? Maybe not weird, just different.

Reply:
It's not "anywhere", but it's not at the root. For example, this is a "model" directory with an accompanying `manifest.json` at the root.

```
model
model/1
model/1/variables
model/1/variables/variables.data-00000-of-00001
model/1/variables/variables.index
model/1/saved_model.pb
model/manifest.json
```